apis/nucleus/src/components/Supernova.jsx

```diff
@@ -119,11 +119,55 @@ function Supernova({ sn, snOptions: options, snPlugins: plugins, layout, appLayo
     keyboardNavigation,
   ]);
 
+  const imageSizingToCssProperty = {
+    originalSize: 'auto auto',
+    alwaysFit: 'contain',
+    fitWidth: '100% auto',
+    fitHeight: 'auto 100%',
+    stretchFit: '100% 100%',
+    alwaysFill: 'cover',
+  };
+
+  function getBackgroundPosition(data) {
+    let bkgImagePosition = 'center center';
+    if (data && data.components && data.components.length > 0 && data.components !== 'none') {
+      const styling = data.components.find((comp) => comp.key === 'general');
```

Comment:
again, this "finding" of the component should be done once per layout change.

Reply:
ok, will change this